json-schema/schema.json

```diff
@@ -50,51 +48,140 @@
           "const": "Collection"
         },
         "assets": {
-          "type": "object",
-          "additionalProperties": {
-            "$ref": "#/definitions/fields"
-          }
+          "$ref": "#/definitions/assets"
         },
         "item_assets": {
-          "type": "object",
-          "additionalProperties": {
-            "$ref": "#/definitions/fields"
-          }
+          "$ref": "#/definitions/assets"
+        },
+        "links": {
+          "$ref": "#/definitions/links"
         }
-      }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/schemes_field"
+        }
+      ]
+    },
+    {
+      "$comment": "This is the schema for STAC Catalogs",
+      "type": "object",
+      "required": [
+        "type"
+      ],
+      "properties": {
+        "type": {
+          "const": "Catalog"
+        },
+        "links": {
+          "$ref": "#/definitions/links"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/schemes_field"
+        }
+      ]
     }
   ],  
   "definitions": {
-    "fields": {
+    "schemes_field": {
       "type": "object",
+      "required": [
+        "storage:schemes"
+      ],
       "properties": {
-        "storage:platform": {
-          "title": "Platform",
-          "type": "string",
-          "enum": [
-            "OTHER",
-            "AWS",
-            "GCP",
-            "AZURE",
-            "IBM",
-            "ALIBABA",
-            "ORACLE"
-          ],
-          "default": "OTHER"
-        },
-        "storage:region": {
-          "title": "Region",
-          "type": "string"
-        },
-        "storage:requester_pays": {
-          "type": "boolean",
-          "title": "Requester pays",
-          "default": false
-        },
-        "storage:tier": {
-          "title": "Tier",
-          "type": "string"
+        "storage:schemes": {
+          "type": "object",
+          "patternProperties": {
+            "^.{1,}$": {
```

Comment:
Yeah, JSON keys can hold anything. My aim was just to prevent empty strings.
If people want we can restrict it, but there's no real technical need for it.
Alternate assets is even worse and uses the same concept, it even allows empty strings.
In this case we should probably also change it there, but it would be breaking....